VMBackup/main/handle_host_daemon.py

```diff
@@ -0,0 +1,861 @@
+#!/usr/bin/env python
+
+import time
+import os
+import threading
+import signal
+import sys
+import json
+from Utils.WAAgentUtil import waagent
+from Utils import HandlerUtil
+import datetime
+from common import CommonVariables
+import subprocess
+import traceback
+from datetime import datetime
+
+IS_PYTHON3 = sys.version_info[0] == 3
+if IS_PYTHON3:
+    import configparser as ConfigParsers
+else:
+    import ConfigParser as ConfigParsers
+
+if IS_PYTHON3:
+    from urllib import request
+else:
+    import urllib2 as request
+
+if IS_PYTHON3:
+    from urllib.error import HTTPError
+else:
+    from urllib2 import HTTPError
+
+if IS_PYTHON3:
+    import urllib.parse as urllib
+else:
+    import urllib
+
+
+SCRIPT_DIR=os.path.dirname(os.path.realpath(__file__))
+BASE_URI="http://168.63.129.16"
+STORAGE_DEVICE_PATH = '/sys/bus/vmbus/devices/'
+GEN2_DEVICE_ID = 'f8b3781a-1e82-4818-a1c3-63d806ec15bb'
```

Comment:
any reason for a hard coded device id?

Reply:
Yes. The resource disk seems to have a fixed GUID when attached to the VM. The old implementation uses this as well [here](https://github.com/Azure/azure-linux-extensions/blob/e6dc7a79bf29f317beeafec29206666f0cec7f0e/VMBackup/main/Utils/ResourceDiskUtil.py#L13C1-L13C56).